src/pages/api/connection/db.ts

```diff
@@ -10,7 +11,7 @@ const handler = async (req: NextApiRequest, res: NextApiResponse) => {
     return res.status(405).json([]);
   }
 
-  const connection = req.body.connection as Connection;
+  const connection = changeTiDBConnectionToMySQL(req.body.connection as Connection);
```

Comment:
The logic seems a bit odd. Can we first check if the connection type is `tidb` and do the transform if it is?

Reply:
Now I check it in the`changeTiDBConnectionToMySQL` function. I will move this logic out of this function.